spatial-partition/src/main/java/com/iluwatar/spatialpartition/QuadTree.java

```diff
@@ -49,33 +52,54 @@
     this.divided = false;
     this.points = new Hashtable<Integer, Point>();
     this.northwest = null; 
-    this.northeast = null; 
+    this.northeast = null;
     this.southwest = null; 
     this.southeast = null;
+    localCoveredBranches = new boolean[8];
   }
 
   void insert(Point p) {
     if (!this.boundary.contains(p)) {
+      // Counting if
+      coveredBranches[0] = true;
+      localCoveredBranches[0] = true;
+      // Counting return
+      coveredBranches[7] = true;
+      localCoveredBranches[7] = true;
```

Comment:
It would be better to just update the local array and to update the "coveredBranches" in the test class with an `@AfterEach`.